glass/src/lib/native/cpp/other/Mechanism2D.cpp

```diff
@@ -90,6 +91,12 @@ class BackgroundInfo {
 BackgroundInfo::BackgroundInfo(Storage& storage)
     : m_filename{storage.GetString("image")} {}
 
+BackgroundInfo::~BackgroundInfo() {
+  if (m_fileOpener) {
+    m_fileOpener->kill();
```

Comment:
Is there a reason kill() isn’t called by the PFD object destructor?  That would avoid all these per-use checks.

Reply:
Nope, fixed